command/server.go

```diff
@@ -286,6 +286,22 @@ func (c *ServerCommand) Run(args []string) int {
 		}
 	}
 
+	// If the backend supports service discovery, run service discovery
+	if coreConfig.HAPhysical != nil {
+		sd, ok := coreConfig.HAPhysical.(physical.ServiceDiscovery)
+		if ok {
+			if err := sd.UpdateAdvertiseAddr(coreConfig.AdvertiseAddr); err != nil {
```

Comment:
We need to make sure that this only takes effect if `advertise_addr` wasn't manually set in the config file.